broker/src/main/java/io/zeebe/broker/Broker.java

```diff
@@ -73,12 +88,30 @@
 public final class Broker implements AutoCloseable {
 
   public static final Logger LOG = Loggers.SYSTEM_LOGGER;
-
+  private static final List<Component<?>> LEADER_COMPONENTS =
+      List.of(
+          new LogStreamComponent(),
+          new RaftLogReaderComponent(),
+          new SnapshotReplicationComponent(),
+          new StateControllerComponent(),
+          new LogDeletionComponent(),
+          new LeaderPostStorageComponent(),
+          new ZeebeDbComponent(),
+          new StreamProcessorComponent(),
+          new SnapshotDirectorComponent(),
+          new RocksDbMetricExporterComponent(),
+          new ExporterDirectorComponent());
+  private static final List<Component<?>> FOLLOWER_COMPONENTS =
+      List.of(
+          new RaftLogReaderComponent(),
+          new SnapshotReplicationComponent(),
+          new StateControllerComponent(),
+          new LogDeletionComponent(),
+          new FollowerPostStorageComponent());
```

Comment:
I think the lists are more closely related to the Broker than the PartitionTransitionImpl itself, no? The transition behavior should be independent of what specific things we open and close. We can have a constructor which uses these lists if none are passed but even then I don't think they should be there at all. The broker already knows (and needs to know) what needs to be opened/closed since it was already passing the relevant data into the ZeebePartition. WDYT?

Reply:
I agree that the lists don't need to be in the `PartitionTransitionImpl` class. But maybe, in a separate class (e.g. `PartitionSteps`). 

My main concern is that the `Broker` class doesn't use the lists itself (separation of concerns) and it is already big and contains a lot of stuff.